src/main/java/org/symphonyoss/symphony/messageml/elements/RoomSelector.java

```diff
@@ -0,0 +1,231 @@
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.exceptions.ProcessingException;
+import org.symphonyoss.symphony.messageml.markdown.nodes.form.PersonSelectorNode;
+import org.symphonyoss.symphony.messageml.markdown.nodes.form.RoomSelectorNode;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class representing a room selector inside a Symphony Elements form.
+ *
+ * @author Mohamed Rojbeni
+ * @since 09/14/2023
+ */
+public class RoomSelector extends FormElement implements LabelableElement, TooltipableElement {
+  public static final String MESSAGEML_TAG = "room-selector";
+
+  private static final String PLACEHOLDER_ATTR = "placeholder";
+  private static final String REQUIRED_ATTR = "required";
+  private static final Set<String> VALID_VALUES_FOR_REQUIRED_ATTR =
+      new HashSet<>(Arrays.asList("true", "false"));
+  private static final String VALUE_ATTR = "value";
+
+  private static final String PRESENTATIONML_TAG = "div";
+  private static final String PRESENTATIONML_PLACEHOLDER_ATTR = "data-placeholder";
+  private static final String PRESENTATIONML_REQUIRED_ATTR = "data-required";
+  private static final String PRESENTATIONML_VALUE_ATTR = "data-value";
+
+  private final static String MARKDOWN = "Room Selector";
+  private static final String CLASS_ATTR = "class";
+  private static final String PRESENTATIONML_NAME_ATTR = "data-name";
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  public RoomSelector(Element parent, FormatEnum messageFormat) {
+    super(parent, MESSAGEML_TAG, messageFormat);
+  }
+
+  @Override
+  public void buildAll(MessageMLParser parser, org.w3c.dom.Element element)
+      throws InvalidInputException, ProcessingException {
+    switch (getFormat()) {
+      case MESSAGEML:
+        super.buildAll(parser, element);
+        break;
+      case PRESENTATIONML:
+        buildElementFromDiv(parser, element);
+        this.validate();
+        break;
+      default:
+        throw new InvalidInputException(
+            String.format("Invalid message format for \"%s\" element", MESSAGEML_TAG));
+    }
+  }
+
+  @Override
+  public void validate() throws InvalidInputException {
+    super.validate();
+
+    if (getAttribute(NAME_ATTR) == null) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
+    if (getAttribute(REQUIRED_ATTR) != null) {
+      assertAttributeValue(REQUIRED_ATTR, VALID_VALUES_FOR_REQUIRED_ATTR);
+    }
+
+    if (getAttribute(VALUE_ATTR) != null) {
+      try {
+        MAPPER.readValue(getAttribute(VALUE_ATTR),
+            MAPPER.getTypeFactory().constructCollectionType(List.class, Long.class));
```

Comment:
This has to be adapted : room references (streamId / threadId) are Strings, not Longs